Add Publish Unit Test Results to GH Actions

artembilan opened the pull request:
* Upgrade dependencies
* Fix repos to not use superset to fail for
non-authorized artifacts

<!--
Thanks for contributing to Spring Integration. 
Please provide a brief description of your pull-request and reference any related issue numbers (prefix references with #) or StackOverflow questions.

See the [Contributor Guidelines for more information](https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc).
-->

artembilan committed:
Add Publish Unit Test Results to GH Actions

* Upgrade dependencies
* Fix repos to not use superset to fail for
non-authorized artifacts

garyrussell commented on .github/workflows/pr-build-workflow.yml:
```diff
@@ -44,3 +44,8 @@ jobs:
+      - name: Publish Unit Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        with:
+          files: '**/build/test-results/**/*.xml'
```
I think capturing the gradle HTML test result rendering is much better than trying to dig through all the XML files in test-results.

artembilan replied:
I'm trying a different action: https://github.com/marketplace/actions/publish-unit-test-results

garyrussell replied:
But I am not averse to capturing both.

artembilan committed:
* Revert Dokka plugin - too much JCenter deps
* Add `if: always()` for Test Results in GH Actions

artembilan committed:
Add `https://repo.spring.io/release` for docs

artembilan committed:
* Use `upload-artifact` action instead -
the `publish-unit-test-result-action` doesn't capture against PRs:
https://github.com/EnricoMi/publish-unit-test-result-action#support-fork-repositories

artembilan committed:
* `if: always()` to be sure that we got an artifact

artembilan committed:
Revert "* `if: always()` to be sure that we got an artifact"

This reverts commit d9d43758405f06a84e936bf41bfb53a7bde64fad.

garyrussell commented on .github/workflows/pr-build-workflow.yml:
```diff
@@ -44,3 +44,11 @@ jobs:
+        with:
+          name: test-results
+          path: '*/build/reports/tests/**/*.*'
+          retention-days: 1
```
I chose 3 days for the other projects so that the artifacts would still be there Monday for PR submitted late Friday.

artembilan committed:
* `retention-days: 3`
